packages/dataviews/src/view-table.tsx

```diff
@@ -92,9 +147,9 @@ const HeaderMenu = forwardRef( function HeaderMenu(
 					variant="tertiary"
 				>
 					{ field.header }
-					{ isSorted && (
+					{ view.sort && isSorted && (
```

Comment:
Here, even if isSorted is actually enough to ensure view.sort is defined I had to add the extra check to make typescript happy. I wonder if there are other native typescript approaches to do that.

Reply:
I see, I think we should do the first option for now, and maybe make everything readonly in a separate PR.